swo/mpt/extensions/core/security.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from typing import Any, Callable, Mapping
+
+import jwt
+from django.http import HttpRequest
+from ninja.security import HttpBearer
+
+logger = logging.getLogger(__name__)
+
+
+class JWTAuth(HttpBearer):
+    JWT_ALGOS = ["HS256"]
+
+    def __init__(
+        self,
+        secret_callback: Callable[[Mapping[str, Any]], str],
+    ) -> None:
+        self.secret_callback = secret_callback
+        super().__init__()
+
+    def authenticate(self, request: HttpRequest, token: str) -> Any | None:
+        audience = request.get_host()
+        try:
+            claims = jwt.decode(
+                token,
+                audience=audience,
+                options={"verify_signature": False},
```

Comment:
Are you sure here?

Reply:
Yes, the idea is the following:
* Decode the token to get the claims (no signature verification)
* Call the jwt_secret_callback with claims, claims contain the webhook id so the callback can return the corresponding secret
* Decode with the secret (it will verify signature, expiration, nbf and audience)